scripts/chgres_cube.py

```diff
@@ -0,0 +1,204 @@
+    external_config_fhrs = external_config["external_model_fhrs"]
+
+    if "task_make_ics" in key_path:
+        if input_type == "grib2":
+            os.environ["fn_grib2"] = external_config_fns[0]
+        else:
+            os.environ["fn_atm"] = external_config_fns[0]
+            os.environ["fn_sfc"] = external_config_fns[1]
+
+        chgres_cube_driver = ChgresCube(
+            config=config_file,
+            cycle=cycle,
+            key_path=key_path,
+            leadtime=dt.timedelta(hours=0),
+        )
+        rundir = Path(chgres_cube_driver.config["rundir"])
+        logging.info(f"Will run in {rundir}")
+        chgres_cube_driver.run()
+
+        if not (rundir / "runscript.chgres_cube.done").is_file():
+            logging.error("Error occurred running chgres_cube. See component error logs.")
+            sys.exit(1)
+
+        # Deliver output data to the forecast's INPUT dir
+        delivery_dir = rundir.parent / "INPUT"
+        _deliver_files(config=expt_config, dst_dir=delivery_dir, key_path=key_path, src_dir=rundir)
+
+    else: # loop over make_lbcs tasks
+        # This loop will need a version of the config that is not dereferenced
+        expt_config = get_yaml_config(config_file)
+
+        fhrs_and_fns = list(zip(external_config_fhrs, external_config_fns))
+        for external_fhr, external_fn in fhrs_and_fns:
+            if input_type == "grib2":
+                os.environ["fn_grib2"] = external_fn
+            else:
+                os.environ["fn_atm"] = external_fn
+
+            lbc_offset_fhrs = expt_config["task_get_extrn_lbcs"]["envvars"][
+                    "EXTRN_MDL_LBCS_OFFSET_HRS"
+                ]
+            fcst_hr_lam = int(external_fhr) - int(lbc_offset_fhrs)
+            leadtime = dt.timedelta(hours=fcst_hr_lam)
+
+            chgres_cube_driver = ChgresCube(
+                config=config_file,
+                cycle=cycle,
+                key_path=key_path,
+                leadtime=leadtime,
+            )
+            rundir = Path(chgres_cube_driver.config["rundir"])
+            logging.info(f"Will run in {rundir}")
+            chgres_cube_driver.run()
+
+            if not (rundir / "runscript.chgres_cube.done").is_file():
+                logging.error("Error occurred running chgres_cube. See component error logs.")
+                sys.exit(1)
+
+            # Use a copy of the original here to avoid opening the file every time.
+            expt_config_cp = get_yaml_config(deepcopy(expt_config.data))
+
+            # output files should contain cycle information relative to the start time of the
+            # forecast -- use cycle = cycle
+            # This dereferencing must be inside loop bc the fcst hour is set differently each time
+            expt_config_cp.dereference(
+                context={
+                    "cycle": cycle,
+                    "leadtime": leadtime,
+                    **os.environ,
+                    **expt_config_cp,
```

Comment:
As mentioned above, it might be worth considering the priority order of the items in this `dict`: Later items will overwrite earlier ones.